### DISCUSSION TITLE
class MyBundle extends AbstractBundle implements CompilerPassInterface

### DISCUSSION THREAD

**Discussion by ahundiak** [symfony/symfony#47687] [Q&A]
I was switching over to the fairly new AbstractBundle class and needed a compiler pass.  Rather than create a new class I used the same trick that Kernel uses and had the bundle class implement the pass interface.  But I still need to manually add the pass:
```
class CeradMyBundle extends AbstractBundle implements CompilerPassInterface
{
  public function build(ContainerBuilder $container): void
  {
      $container->addCompilerPass($this); // Can this be automated???
  }
  public function process(ContainerBuilder $container) // the pass
  {
```
Not a big deal but, in the interest of consistency or whatever, I wonder if there could be a check to see if a bundle class implements the interface and then automatically add it?

Or maybe there is a reason why having the bundle implement the interface is a bad idea even though it seems to work?

**Comment by MatTheCat** [comment]
I’m not aware of any restriction preventing this :thinking: 

You could try submitting a PR updating [`Kernel::prepareContainer`](https://github.com/symfony/symfony/blob/4c6049e9b432f511b6b7c32eb495e2dfeeee43e4/src/Symfony/Component/HttpKernel/Kernel.php#L595) and see how it goes!

**Reply by ahundiak** [reply]
Thanks.  Last time I tried making a PR it did not end well.  But try try again.

**Reply by MatTheCat** [reply]
Someone did it for you: https://github.com/symfony/symfony/pull/62800 (to be released with Symfony 8.1).
